src/plan/generational/global.rs

```diff
@@ -298,21 +300,53 @@ impl<VM: VMBinding> CommonGenPlan<VM> {
     }
 }
 
-/// This trait include methods that are specific to generational plans.
+/// This trait includes methods that are specific to generational plans. This trait needs
+/// to be object safe.
 pub trait GenerationalPlan: Plan {
-    /// Return the common generational implementation [`crate::plan::generational::global::CommonGenPlan`].
-    fn common_gen(&self) -> &CommonGenPlan<Self::VM>;
+    /// Is the current GC a nursery GC? If a GC is not a nursery GC, it will be a full heap GC.
+    /// This should only be called during GC.
+    fn is_current_gc_nursery(&self) -> bool;
+
+    /// Is the object in the nursery?
+    fn is_object_in_nursery(&self, object: ObjectReference) -> bool;
```

Comment:
This function only works based on the assumption that we only have two generations in the heap (i.e. young-gen and old-gen). If the number of generations is larger than 2, shall we do something different to handle this?

As an example, G1 has three generations: eden, survived, and old.

Reply:
If we need to support multiple generations, we will have to do some refactoring to allow it. For example, `gen_plan.is_object_in_nursery(object, Generationa::Eden)`. Alternatively, we may use a new trait, like `MultiGenerationalPlan`. I don't know what would be a better way.

The changes in this PR is to allow us to reuse the generational code for sticky plans. I think if we want to support multiple generations, we would also want to reuse the current generational code as much as we can.